spec/tapioca/dsl/compilers/json_api_client_resource_spec.rb

```diff
@@ -368,6 +368,73 @@ def tag_count=(tag_count); end
 
                 assert_equal(expected, rbi_for(:Post))
               end
+
+              it "prints a warning that #sorbet_type is deprecated" do
+                add_ruby_file("post.rb", <<~RUBY)
+                  class CustomType
+                    def self.sorbet_type
+                      "Integer"
+                    end
+                  end
+
+                  class Post < JsonApiClient::Resource
+                    property :comment_count, type: :custom_type
+                    property :tag_count, type: :custom_type, default: 0
+                  end
+                RUBY
+
+                expected = <<~RBI
+                  # typed: strong
+
+                  class Post
+                    include JsonApiClientResourceGeneratedMethods
+
+                    module JsonApiClientResourceGeneratedMethods
+                      sig { returns(T.nilable(Integer)) }
+                      def comment_count; end
+
+                      sig { params(comment_count: T.nilable(Integer)).returns(T.nilable(Integer)) }
+                      def comment_count=(comment_count); end
+
+                      sig { returns(Integer) }
+                      def tag_count; end
+
+                      sig { params(tag_count: Integer).returns(Integer) }
+                      def tag_count=(tag_count); end
+                    end
+                  end
+                RBI
+
+                assert_output(nil, /WARNING: `#sorbet_type` is deprecated./) do
+                  assert_equal(expected, rbi_for(:Post))
+                end
+              end
+
+              it "does not crash if the source location #sorbet_type is unknown" do
+                add_ruby_file("post.rb", <<~RUBY)
+                  class CustomType
+                    def self.sorbet_type
+                      "Integer"
+                    end
+
+                    def self.method(name)
+                      m = super
+                      # Fake the source location, as if this was defined in a C extension.
+                      def m.source_location
```

Comment:
It's surprisingly hard to make a method with no `source_location` ahha. Native methods from C extensions would have it, but other than that, every way to define a Ruby method ended up having it defined. I tried:

1. `define_singleton_method`
2. `define_method`
3. `eval`
4. `class_eval`
5. `Class.new { ... }`

and they all had it defined.

Reply:
Interesting.

Reply:
IKR? I could have sworn you needed to explicitly pass in the `___FILE__` and `__LINE__` else you got no info at all